### ISSUE TITLE
Question about TTA on Semantic Segmentation

### ISSUE THREAD

**Issue by chengkaiAcademyCity** [mihirp1998/Diffusion-TTA#1]
Thanks for your wonderful work! I see your paper's results on TTA on Semantic Segmentation. I am wondering how you do Backprop to Segmentation Conditions. Will that part of the code be released?

<img width="993" alt="Screenshot 2023-12-02 at 11 36 45 PM" src="https://github.com/mihirp1998/Diffusion-TTA/assets/73343659/6de6412a-c1ad-455b-aaae-7bd2dbee1a42">

**Comment by buttomnutstoast** [comment]
Thanks for you interest in our work!

For conditioning on semantic segmentation and depth estimation, we concatenate the predicted segmentation and depth map with the noisy image latent.  Please see Figure 7 in [our paper](https://arxiv.org/abs/2311.16102).  The segmenter and the depth estimator are updated via back propagating gradients through the prediction.

We are still considering whether to release the code for TTA on segmentation and depth estimation.  Integration to the current code base is non-trivial since re-training the diffusion generative model is required.

**Comment by sfengpeng** [comment]
> Thanks for you interest in our work!
> 
> For conditioning on semantic segmentation and depth estimation, we concatenate the predicted segmentation and depth map with the noisy image latent. Please see Figure 7 in [our paper](https://arxiv.org/abs/2311.16102). The segmenter and the depth estimator are updated via back propagating gradients through the prediction.
> 
> We are still considering whether to release the code for TTA on segmentation and depth estimation. Integration to the current code base is non-trivial since re-training the diffusion generative model is required.

Thanks for your work! I also hope the code part of Semantic Segmentation to be realeased!

**Comment by zenghy96** [comment]
Thanks for your wonderful work!
Have you tried the TTA segmentation task on other datasets, such as adapting GTA5 to Cityscapes?
